src/app/chat/chat.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ChatService } from './chat.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-chat',
+  templateUrl: './chat.component.html',
+  styleUrls: ['./chat.component.css'],
+})
+export class ChatComponent implements OnInit, OnDestroy {
+  message = '';
+  messages: string[] = [];
+  sub: Subscription;
+
+  constructor(private chatService: ChatService) {}
+
+  sendMessage() {
+    this.chatService.sendMessage(this.message);
+    this.message = '';
+  }
+ 
+  ngOnInit() {
+    this.sub = this.chatService.getMessages().subscribe((message: string) => {
+      this.messages.push(message);
+    });
```

Comment:
As I remember, you using effects. Please use effects across all components of your application for work with services

Reply:
ok